make/tests

```diff
@@ -38,16 +38,28 @@ stampy "${METRICS}" "$0" "make-tests::${POD_NAME}::setup" end
 
 stampy "${METRICS}" "$0" "make-tests::${POD_NAME}::create" start
 
+# Replace the fixed secret in the relevant task definition with the
+# actual name as pulled from the cluster under test.
+
+capsecret="$(kubectl get pod api-0 --namespace "${NAMESPACE}" -o jsonpath='{@.spec.containers[0].env[?(@.name=="MONIT_PASSWORD")].valueFrom.secretKeyRef.name}')"
+#echo cap-secret = $capsecret
+
+remove_test_config () { rm test.yaml ; }
+sed < "output/kube/bosh-task/${POD_NAME}.yaml" \
+    > test.yaml \
+    "s|name: \"secret\"|name: \"${capsecret}\"|"
+trap remove_test_config EXIT
+
 # Delete left-over pod/definition from previous runs, then create/run
-kubectl delete --namespace="${NAMESPACE}" --filename="output/kube/bosh-task/${POD_NAME}.yaml" \
+kubectl delete --namespace="${NAMESPACE}" --filename="test.yaml" \
```

Comment:
We should be able to delete with the old file with the incorrect secrets (that's not used to match what to delete, only namespace/name I think).  Otherwise this wouldn't work if you upgraded CF in the middle...

Reply:
Wait, you are talking about running an upgrade while the test are running ?
Is that something we have ever supported ?